### ISSUE TITLE
Slidee width calculation issue

### ISSUE THREAD

**Issue by lolo34140** [darsain/sly#200]
Hello,

I observed a problem with the slidee width caluclation.
The width of Slidee if different from the total of item width...

The main issue is about page bullets and prev next buttons that show activated or two much bullets.
The issue depend of the browser width and floating number.

example 1: http://jsfiddle.net/pj29P/30/ (in principle 2 page bullets depends of your browser width)
example 2: http://jsfiddle.net/pj29P/40/ (in principle 1 page bullet depends of your browser width)

Lolo

**Comment by darsain** [comment]
I really appreciate the fiddles, unfortunately they don't work for me due to mixed content policies (jsfiddle is https and your external resources are http, so they refuse to load).

Also, could you just paste the minimal needed code to replicate the issue? And not load external CSS files and plugins? Just style it as basically as possible. Also the plugins file has only jquery easing, which is redundant if you leave the `easing` option to default (best for demos).

For sly, use some of the CDN links, as github pages don't do https. Like this one:

```
//cdn.jsdelivr.net/sly/1.5.1/sly.min.js
```

Sorry for the hassle.

**Comment by lolo34140** [comment]
The demo is based from one of you fiddle....

**Comment by darsain** [comment]
Yeah I know, and I'm sorry about that :) It doesn't work anymore due to problems mentioned above. I think jsfiddle wasn't forcing https back than.

**Comment by lolo34140** [comment]
http://jsfiddle.net/pj29P/46/

**Comment by darsain** [comment]
Thx. I'll look at it when I'll have some time.

**Comment by 2ramiro** [comment]
Hello
I think i have the same problem , the width calculated from sly is not enough so the last item goes into the next line.
Here is a Link to my test site where you can see the issue:
http://test.kultur.lv/Produkte/asc/alternativ-toner-fuer-oki-b-250025202540okifax-2510okioffice-25102530-hohe-kapazitaet-09004391/3950159
Thanks 
Ramiro

**Comment by serafimsaudade8** [comment]
Did a Pull Request to fix this issue

**Comment by critcat** [comment]
Yes, there is a problem. When not specified width for each element in the horizontal mode, the width is calculated incorrectly.

As you can see, when you do not set the width value in the css for .frame ul li, the last element is transferred to the new line: https://jsfiddle.net/Lp3hkrms/1/

However, when the width is set, all works: https://jsfiddle.net/16tzrapb/1/

**Comment by dannevesdantas** [comment]
I have found a simple workaround for this issue:

```javascript
    // fix for incorrect sly.js width calculation
    // https://github.com/darsain/sly/issues/200
    $(".frame ul li").each(function(index, element) {
        var itemWidth = $(this).outerWidth(); // get item's width
        $(this).css("width", itemWidth); // set item's width in css
    });
```

Working example at [https://jsfiddle.net/dannevesdantas/79vqy0kL/](https://jsfiddle.net/dannevesdantas/79vqy0kL/)

**Comment by Azat74** [comment]
http://prntscr.com/jx194z oh wait need +1px
